### PR TITLE
New VK Backend

### PR DESCRIPTION
**Pull request by sargun**
This VK Backend has a few changes:

* It no longer uses pipes. Instead, it just drops the status
  as a flat file on disk
* It moves the systemd unit for initialization of the titus executor
  into titus-executor repo (this sheds some stuff)

### CODE REVIEW

**Comment by sargun** [comment]
@solarkennedy Can you make sure this doesn't interfere with your storage work? Can you add detach / attach bits as drop-in extensions?

**Comment by solarkennedy** [comment]
So the process tree will be:
```
vk ->
  vk-backend ? ->
    titus-executor ->
      docker stuff or whatever
```
Why though? How does this save systemd any burden?

**Comment by sargun** [comment]
vk-backend is titus-executor -- just a different API to titus-executor. It uses flat files to pass state back and forth instead of unix pipes.

**Comment by sargun** [comment]
This does introduce one kind of big change, which is if you're killing titus-executors on the host, you have to run:

```
systemctl kill --kill-who=main titus-executor@...
```

We can work around this by having the VK originate its own status if the unit is killed in an unexpected fashion. Or add an ExecStopPost which holds out for 10 seconds or something to try to have the VK pickup the state.

**Review by rgulewich — APPROVED** [review]

File: root/lib/systemd/system/titus-executor-backend@.service [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=Titus Virtual Kubelet Executor Service
+
+[Service]
+Slice=containers.slice
+ExecStartPre=/bin/mkdir -p /run/titus-executor/pod-status
+ExecStart=/apps/titus-executor/bin/run-titus-virtual-kubelet-backend.sh --runtime-dir=/run/titus-executor/%i
+Restart=no
```
Comment (sargun) [inline root/lib/systemd/system/titus-executor-backend@.service:8]: curl http://localhost:vk/foo -d @/run/titus-executor/%i/statejson

**Review by alekar — APPROVED** [review]

**Review by solarkennedy — APPROVED** [review]
I would like to know what your plan is with regards to rolling this out immutably.

**Pull request merged**